packages/design/Icon/Icon.tsx

```diff
@@ -0,0 +1,11 @@
+import type { IconNames } from "./types";
+
+export type IconProps = { name: IconNames; size?: number; className: string };
+
+export function Icon({ name, size = 20, className }: IconProps) {
+  return (
+    <svg width={size} height={size} viewBox="0 0 24 24" fill="currentColor" className={className}>
+      <use href={`/icons/sprite.svg#${name}`} />
```

Comment:
I don't understand how this (the path `/icons/sprite.svg#${name}`) works.

The actual icon files are at `packages/design/svgs/<name>.svg` and there is also a `public/icons/sprite.svg` that seems to contain a bunch of the path data that's...duplicate?

For example, `comments.svg`:
```html
<svg
  xmlns="http://www.w3.org/2000/svg"
  width="24"
  height="24"
  viewBox="0 0 24 24"
  fill="none"
>
  <g id="comments">
    <path
      id="path"
      d="M3.70711 15.2929C3.07714 15.9229 2 15.4767 2 14.5858V3C2 2.71667 2.096 2.479 2.288 2.287C2.47933 2.09567 2.71667 2 3 2H16C16.2833 2 16.5207 2.09567 16.712 2.287C16.904 2.479 17 2.71667 17 3V12C17 12.2833 16.904 12.5207 16.712 12.712C16.5207 12.904 16.2833 13 16 13H6.41421C6.149 13 5.89464 13.1054 5.70711 13.2929L3.70711 15.2929ZM7 18C6.71667 18 6.479 17.904 6.287 17.712C6.09567 17.5207 6 17.2833 6 17V15C6 15 16.8329 15 18 15C18.5523 15 19 14.5523 19 14V6C19.6667 6 20.3333 6 21 6C21.2833 6 21.5207 6.09567 21.712 6.287C21.904 6.479 22 6.71667 22 7V19.5858C22 20.4767 20.9229 20.9229 20.2929 20.2929L18.2929 18.2929C18.1054 18.1054 17.851 18 17.5858 18H7Z"
      fill="black"
    />
  </g>
</svg>
```
Which seems to correspond to this symbol in `sprite.svg`:
```html
<symbol
  xmlns="http://www.w3.org/2000/svg"
  id="comments"
  width="24"
  height="24"
  viewBox="0 0 24 24"
  ><path
    d="M3.70711 15.2929C3.07714 15.9229 2 15.4767 2 14.5858V3C2 2.71667 2.096 2.479 2.288 2.287C2.47933 2.09567 2.71667 2 3 2H16C16.2833 2 16.5207 2.09567 16.712 2.287C16.904 2.479 17 2.71667 17 3V12C17 12.2833 16.904 12.5207 16.712 12.712C16.5207 12.904 16.2833 13 16 13H6.41421C6.149 13 5.89464 13.1054 5.70711 13.2929L3.70711 15.2929ZM7 18C6.71667 18 6.479 17.904 6.287 17.712C6.09567 17.5207 6 17.2833 6 17V15C6 15 16.8329 15 18 15C18.5523 15 19 14.5523 19 14V6C19.6667 6 20.3333 6 21 6C21.2833 6 21.5207 6.09567 21.712 6.287C21.904 6.479 22 6.71667 22 7V19.5858C22 20.4767 20.9229 20.9229 20.2929 20.2929L18.2929 18.2929C18.1054 18.1054 17.851 18 17.5858 18H7Z"
/></symbol>
```
Both have the exact same path data string:
```
M3.70711 15.2929C3.07714 15.9229 2 15.4767 2 14.5858V3C2 2.71667 2.096 2.479 2.288 2.287C2.47933 2.09567 2.71667 2 3 2H16C16.2833 2 16.5207 2.09567 16.712 2.287C16.904 2.479 17 2.71667 17 3V12C17 12.2833 16.904 12.5207 16.712 12.712C16.5207 12.904 16.2833 13 16 13H6.41421C6.149 13 5.89464 13.1054 5.70711 13.2929L3.70711 15.2929ZM7 18C6.71667 18 6.479 17.904 6.287 17.712C6.09567 17.5207 6 17.2833 6 17V15C6 15 16.8329 15 18 15C18.5523 15 19 14.5523 19 14V6C19.6667 6 20.3333 6 21 6C21.2833 6 21.5207 6.09567 21.712 6.287C21.904 6.479 22 6.71667 22 7V19.5858C22 20.4767 20.9229 20.9229 20.2929 20.2929L18.2929 18.2929C18.1054 18.1054 17.851 18 17.5858 18H7Z
```
Why do both exist? (Why not just the `symbol`?) 😅

Reply:
> I don't understand how this (the path /icons/sprite.svg#${name}) works.

[This article](https://css-tricks.com/svg-use-with-external-reference-take-2/) expands on the topic more, but TLDR the `use` tag is a way to instance an SVG for better performance. The `Icon` component reaches into the generated sprite to pull out instances of individual icons.

> Why do both exist? (Why not just the symbol?)

The main reason for the `svgs` directory is to provide an escape hatch if the [API is ever down](https://github.com/replayio/devtools/tree/c0447508c37556ac2de9b1f77052f702ec9f16da/packages/design/Icon#figma-api-down). Another reason prior, was because React didn't output any visuals so the idea was to easily be able to see diffs for individual icons. I think this still holds true and it's worth checking in each icon so reviews and history are easier. I'll detail more in the Icon README so this is clear.